packages/open-payments/src/client/requests.ts

```diff
@@ -158,90 +199,113 @@ const checkUrlProtocol = (deps: BaseDeps, url: string): string => {
   return requestUrl.href
 }
 
-export const createAxiosInstance = (args: {
+type CreateHttpClientArgs = {
   requestTimeoutMs: number
-  privateKey?: KeyObject
-  keyId?: string
-}): AxiosInstance => {
-  const axiosInstance = axios.create({
+} & (
+  | { privateKey?: KeyObject; keyId?: string }
+  | { authenticatedRequestInterceptor: InterceptorFn }
+)
+
+export type HttpClient = KyInstance
+
+export type InterceptorFn = (request: Request) => Request | Promise<Request>
+
+export const createHttpClient = async (
+  args: CreateHttpClientArgs
+): Promise<HttpClient> => {
+  const { default: ky } = await import('ky')
+
+  const kyInstance = ky.create({
+    timeout: args?.requestTimeoutMs,
     headers: {
-      common: {
-        Accept: 'application/json',
-        'Content-Type': 'application/json'
-      }
-    },
-    timeout: args.requestTimeoutMs
+      Accept: 'application/json',
+      'Content-Type': 'application/json'
+    }
   })
 
-  if (args.privateKey !== undefined && args.keyId !== undefined) {
-    const privateKey = args.privateKey
-    const keyId = args.keyId
-    axiosInstance.interceptors.request.use(
-      async (config: InternalAxiosRequestConfig) => {
-        if (!config.method || !config.url) {
-          throw new Error('Cannot intercept request: url or method missing')
-        }
-        const contentAndSigHeaders = await createHeaders({
-          request: {
-            method: config.method.toUpperCase(),
-            url: config.url,
-            headers: JSON.parse(JSON.stringify(config.headers)),
-            body: config.data ? JSON.stringify(config.data) : undefined
-          },
-          privateKey,
-          keyId
-        })
-        if (config.data) {
-          config.headers['Content-Digest'] =
-            contentAndSigHeaders['Content-Digest']
-          config.headers['Content-Length'] =
-            contentAndSigHeaders['Content-Length']
-          config.headers['Content-Type'] = contentAndSigHeaders['Content-Type']
-        }
-        config.headers['Signature'] = contentAndSigHeaders['Signature']
-        config.headers['Signature-Input'] =
-          contentAndSigHeaders['Signature-Input']
-        return config
-      },
-      undefined,
-      {
-        runWhen: (config: InternalAxiosRequestConfig) =>
-          config.method?.toLowerCase() === 'post' ||
-          !!(config.headers && config.headers['Authorization'])
+  let requestInterceptor: InterceptorFn | undefined
+
+  if ('authenticatedRequestInterceptor' in args) {
+    requestInterceptor = (request) => {
+      if (requestShouldBeAuthorized(request)) {
+        return args.authenticatedRequestInterceptor(request)
       }
-    )
+
+      return request
+    }
+  } else if ('privateKey' in args && 'keyId' in args) {
```

Comment:
feel free to disregard if you like it as-is, but figured i'd leave this suggestion for convenience

```suggestion
  } else {
```

Reply:
Makes sense, we also do a check at runtime when initializing the client anyway :+1: